ui.apps/src/main/content/jcr_root/apps/core/fd/components/aemform/v1/aemform/aemform.html

```diff
@@ -42,10 +42,17 @@
 <sly data-sly-use.clientLib="${'/libs/granite/sightly/templates/clientlib.html'}"/>
 <sly data-sly-test.useIframe="${form.useIframe != 'false'}">
     <sly data-sly-test.formLocaleString="${form.locale || 'en'}"/>
-    <iframe class="cmp-aemform__iframecontent"
-            src="${request.contextPath}${resource.path}.iframe.${formLocaleString}.html?dataRef=${request.requestParameterMap['dataRef'] != null ? request.requestParameterMap['dataRef'][0].toString : '' @ context='attribute'}${!wcmmode.edit ? '&wcmmode={0}':'' @ format=[wcmmode.toString]}"
-            stye="width:100%;"
-            data-form-page-path="${form.formEditPagePath}"></iframe>
+    <sly data-sly-set.dataRef="${request.requestParameterMap['dataRef'] != 'unsafe' ? request.requestParameterMap['dataRef'][0].toString : ''}"/>
+    <sly data-sly-set.wcmmode="${!wcmmode.edit ? 'wcmmode={0}' : '' @ format=[wcmmode.toString]}"/>
+
+    <sly data-sly-set.urlParams="${dataRef}${dataRef && wcmmode ? '&' : ''}${wcmmode}"/>
+
+    <sly data-sly-test="${urlParams}">
```

Comment:
Code does not look readable, even if URL params are not present, iframe should still be rendered.

Reply:
+1

@AnurudraS reconfirming - have you validated all the 4 options:
* No dataRef - No wcmmode (no ? or &)
* dataRef and No wcmmode (?)
* No dataRef and wcmmode (?)
* dataRef and wcmmode (? &)

Reply:
@AnurudraS Please try to close this today and raise updated PR..

Reply:
@iamsudhanshu, a small change is done I don't know how it got left. I have updated the code now No dataRef - No wcmmode (no ? or &) case will also be valid there won't be any '?' as well when there is nothing in the urlParams